packages/sdk/src/entities/account.ts

```diff
@@ -61,22 +68,28 @@ export class AccountManager {
 
   // Get the supply source contract
   getSupplySource(
-    client: RecallClient,
-    address?: Address,
+    chain: Chain,
+    contractAddress?: Address,
   ): GetContractReturnType<typeof recallErc20Abi, Client, Address> {
-    const chainId = client.publicClient?.chain?.id;
-    if (!chainId) {
-      throw new Error("Client chain ID not found");
-    }
+    const chainId = chain.id;
     const deployedSupplySourceAddress = (
       recallErc20Address as Record<number, Address>
     )[chainId];
-    if (!deployedSupplySourceAddress) {
+    const overrideConfig =
+      this.client.contractOverrides.accountManager?.recallErc20;
+    const overrideSupplySourceAddress = overrideConfig
+      ? overrideConfig[chainId]
+      : undefined;
+    const override =
+      contractAddress ??
+      overrideSupplySourceAddress ??
+      deployedSupplySourceAddress;
```

Comment:
note: this pattern appears in a couple of places. i didn't want to overhaul the `AccountManager` etc too much, but it's a little messy imo. i'll take another pass in a different PR on refactoring a few other pieces in #79